src/HDF5.jl

```diff
@@ -931,7 +970,18 @@ end
 t_commit(parent::Union{HDF5File, HDF5Group}, path::String, dtype::HDF5Datatype) = t_commit(parent, path, dtype, p_create(H5P_LINK_CREATE))
 
 a_create(parent::Union{HDF5File, HDF5Object}, name::String, dtype::HDF5Datatype, dspace::HDF5Dataspace) = HDF5Attribute(h5a_create(checkvalid(parent).id, name, dtype.id, dspace.id), file(parent))
-p_create(class, toclose=false) = HDF5Properties(h5p_create(class), toclose)
+p_create(class, toclose=false) = HDF5Properties(h5p_create(class), toclose, class)
+function p_create(class, toclose, prop1::String, val1, pv...) 
```

Comment:
The code and API here would be cleaner if we used `pv::Pair{String,<:Any}...`. Also `checkprops` would be much simpler. This is basically unrelated to your PR unless users need to provide properties for MPI use.

Reply:
I thought of that as well, a Pait or even Dict based approach would be more cleaner definitely. But for the same reason as above, I used this approach to not break compatibility of existing routines like d_create, h5open, setindex! ,which are already using a simple parameter list.
This could eventually be changed once the improved keyword argument support in 0.7/1.0 is available.